Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingPass.cpp

```diff
@@ -301,17 +346,22 @@ namespace AZ
 
                 AZStd::shared_ptr<RHI::RayTracingShaderTableDescriptor> descriptor = AZStd::make_shared<RHI::RayTracingShaderTableDescriptor>();
 
-                if (rayTracingFeatureProcessor->GetSubMeshCount())
+                if (rayTracingFeatureProcessor->HasGeometry())
                 {
                     // build the ray tracing shader table descriptor
                     RHI::RayTracingShaderTableDescriptor* descriptorBuild = descriptor->Build(AZ::Name("RayTracingShaderTable"), m_rayTracingPipelineState)
                         ->RayGenerationRecord(AZ::Name(m_passData->m_rayGenerationShaderName.c_str()))
                         ->MissRecord(AZ::Name(m_passData->m_missShaderName.c_str()));
 
-                    // add a hit group for each mesh to the shader table
-                    for (uint32_t i = 0; i < rayTracingFeatureProcessor->GetSubMeshCount(); ++i)
+                    // add a hit group for standard meshes mesh to the shader table
+                    descriptorBuild->HitGroupRecord(AZ::Name("HitGroup"));
+
+                    // add a hit group for each procedural geometry type to the shader table
+                    const auto& proceduralGeometryTypes = rayTracingFeatureProcessor->GetProceduralGeometryTypes();
+                    for (auto it = proceduralGeometryTypes.cbegin(); it != proceduralGeometryTypes.cend(); ++it)
                     {
-                        descriptorBuild->HitGroupRecord(AZ::Name("HitGroup"));
+                        descriptorBuild->HitGroupRecord(it->m_name);
+                        // TODO(intersection): Set per-hitgroup SRG once RayTracingPipelineState supports local root signatures
                     }
```

Comment:
Is this not necessary anymore?

Reply:
Not really clear now. In theory using the Bindless SRG should be enough, but it would be more user friendly to be able to use a PassSRG-like construct.